jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-client/src/main/java/org/jbpm/workbench/cm/client/roles/CaseRolesPresenter.java

```diff
@@ -16,203 +16,214 @@
+                .name(caseRoleAssignmentSummary.getName())
+                .users(caseRoleAssignmentSummary.getUsers())
+                .groups(caseRoleAssignmentSummary.getGroups()).build());
+        editRoleAssignmentView.show(() -> assignToRole());
+    }
+
+    protected CaseRoleAssignmentSummary getRoleAssignment(final CaseInstanceSummary cis, final String roleName) {
+        if (CASE_OWNER_ROLE.equals(roleName)) {
+            return CaseRoleAssignmentSummary.builder().name(roleName).users(Arrays.asList(cis.getOwner())).build();
         }
+        return cis.getRoleAssignments()
+                .stream()
+                .filter(ra -> roleName.equals(ra.getName()))
+                .findFirst().orElse(CaseRoleAssignmentSummary.builder().name(roleName).build());
     }
 
-    protected void addUserToRole(final String userName, final String roleName) {
-        caseService.call(
-                (Void) -> findCaseInstance()
-        ).assignUserToRole(serverTemplateId, containerId, caseId, roleName, userName);
+    protected void assignToRole() {
+        final List<String> assignmentErrors =
+                caseRolesValidations.validateRolesAssignments(caseDefinition,
+                        Arrays.asList(editRoleAssignmentView.getValue()));
+        if (assignmentErrors.isEmpty() == false) {
+            editRoleAssignmentView.showValidationError(assignmentErrors);
+            return;
+        }
+        editRoleAssignmentView.hide();
+        storeRoleAssignments(editRoleAssignmentView.getValue().getName(),
+                editRoleAssignmentView.getValue().getUsers(),
+                editRoleAssignmentView.getValue().getGroups());
     }
 
-    protected void addGroupToRole(final String groupName, final String roleName) {
-        caseService.call(
-                (Void) -> findCaseInstance()
-        ).assignGroupToRole(serverTemplateId, containerId, caseId, roleName, groupName);
+    protected CaseRoleAssignmentSummary getRoleAssignmentByName(String roleName) {
+        return allUnfilteredElements.stream()
+                .filter(caseRoleAssignmentSummary -> roleName.equals(caseRoleAssignmentSummary.getName()))
+                .findFirst().get();
+    }
+
+    protected void storeRoleAssignments(String roleName, List<String> users, List<String> groups) {
+        CaseRoleAssignmentSummary previousCaseRoleAssignmentSummary = getRoleAssignmentByName(roleName);
+        List<String> previousUserAssignments = previousCaseRoleAssignmentSummary.getUsers();
+        List<String> previousGroupsAssignments = previousCaseRoleAssignmentSummary.getGroups();
+        previousCaseRoleAssignmentSummary.setGroups(new ArrayList<>(groups));
+        previousCaseRoleAssignmentSummary.setUsers(new ArrayList<>(users));
+        filterElements();
+
+        List<String> usersToRemove = new ArrayList<String>(previousUserAssignments);
+        usersToRemove.removeAll(users);
+        usersToRemove.forEach(
```

Comment:
@nmirasch I could actually see duplicated user/group names in the demo version at least. I think it could be because of the fact that we're using lists and the converter also doesn't prevent duplicates. Maybe the fix isn't in here specifically but needs a double check.